src/membrain_seg/networks/unet.py

```diff
@@ -0,0 +1,146 @@
+from typing import Dict, Tuple
+
+import pytorch_lightning as pl
+import torch
+from monai.data import decollate_batch
+from monai.inferers import sliding_window_inference
+from monai.losses import DiceFocalLoss
+from monai.metrics import DiceMetric
+from monai.networks.layers import Norm
+from monai.networks.nets import UNet as MonaiUnet
+from monai.transforms import AsDiscrete, Compose, EnsureType
+
+
+class SemanticSegmentationUnet(pl.LightningModule):
+    """Implementation of a Unet for semantic segmentation.
+
+    This is uses the monai Unet. See the monai docs for more details
+    on the parameters.
+    https://docs.monai.io/en/stable/networks.html#unet
+
+    Parameters
+    ----------
+    spatial_dims : int
+        The number of spatial dimensions is in the data.
+    in_channels : int
+        The number of channels in the input tensor.
+        The default value is 1.
+    out_channels : int
+        The number of channels in the output tensor.
+        The default value is 1.
+    channels : Tuple[int, ...]
+        The number of channels in each layer of the encoder/decoder.
+        Default value is (16, 32, 64, 128, 256)
+    strides : Tuple[int, ...]
+        The strides for the convolutions. Must have len(channels - 1) elements.
+        Default value is (2, 2, 2, 2)
+    learning_rate : float
+        The learning rate for the Adam optimizer.
+        Default value is 1e-4.
+    image_key : str
+        The value in the batch data dictionary containing the input image.
+        Default value is "image".
+    label_key : str
+        The value in the batch data dictionary containing the labels.
+        Default value is "label"
+    roi_size : Tuple[int, ...]
+        The size of the sliding window for the validation inference.
+        Default value is (160, 160, 160).
+    """
+
+    def __init__(
+        self,
+        spatial_dims: int = 3,
+        in_channels: int = 1,
+        out_channels: int = 1,
+        channels: Tuple[int, ...] = (16, 32, 64, 128, 256),
+        strides: Tuple[int, ...] = (2, 2, 2, 2),
+        num_res_units=2,
```

Comment:
```suggestion
        num_res_units=2,
```

not in parameters in docstring - not 100% what this refers to without delving into the monai implementation

Reply:
I added a docstring